src/targets/gpu/include/migraphx/gpu/hipblaslt.hpp

```diff
@@ -0,0 +1,109 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#ifndef MIGRAPHX_GUARD_MIGRAPHLIB_HIPBLASLT_HPP
+#define MIGRAPHX_GUARD_MIGRAPHLIB_HIPBLASLT_HPP
+#include <migraphx/argument.hpp>
+#include <migraphx/manage_ptr.hpp>
+#include <migraphx/gpu/config.hpp>
+#include <migraphx/errors.hpp>
+#if MIGRAPHX_USE_HIPBLASLT
+#include <hipblaslt/hipblaslt.h>
+#include <hipblaslt/hipblaslt-ext.hpp>
+
+// TODO: Remove hipblasStatusToString() function when hipblaslt
+// provides an API for doing this in hipBLASLt.
+
+// Convert hipblas_status to string
+inline const char* hipblasStatusToString(hipblasStatus_t status)
+{
+    switch(status)
+    {
+    case HIPBLAS_STATUS_SUCCESS: return "HIPBLAS_STATUS_SUCCESS";
+    case HIPBLAS_STATUS_NOT_INITIALIZED: return "HIPBLAS_STATUS_NOT_INITIALIZED";
+    case HIPBLAS_STATUS_ALLOC_FAILED: return "HIPBLAS_STATUS_ALLOC_FAILED";
+    case HIPBLAS_STATUS_INVALID_VALUE: return "HIPBLAS_STATUS_INVALID_VALUE";
+    case HIPBLAS_STATUS_MAPPING_ERROR: return "HIPBLAS_STATUS_MAPPING_ERROR";
+    case HIPBLAS_STATUS_EXECUTION_FAILED: return "HIPBLAS_STATUS_EXECUTION_FAILED";
+    case HIPBLAS_STATUS_INTERNAL_ERROR: return "HIPBLAS_STATUS_INTERNAL_ERROR";
+    case HIPBLAS_STATUS_NOT_SUPPORTED: return "HIPBLAS_STATUS_NOT_SUPPORTED";
+    case HIPBLAS_STATUS_ARCH_MISMATCH: return "HIPBLAS_STATUS_ARCH_MISMATCH";
+    case HIPBLAS_STATUS_HANDLE_IS_NULLPTR: return "HIPBLAS_STATUS_HANDLE_IS_NULLPTR";
+    case HIPBLAS_STATUS_INVALID_ENUM: return "HIPBLAS_STATUS_INVALID_ENUM";
+    case HIPBLAS_STATUS_UNKNOWN: return "HIPBLAS_STATUS_UNKNOWN";
+    }
+    return "<undefined hipblasStatus_t value>";
+}
+
+template <class F, class... Ts>
+inline auto hipblaslt_invoke(F f, Ts... xs)
+{
+    // Call the function `f` with `xs...` and capture the status
+    auto status = f(xs...);
+
+    if(status != HIPBLAS_STATUS_SUCCESS)
+    {
+        std::string error_message =
+            "hipBLAS error: '" + std::string(hipblasStatusToString(status)) + "'(" +
+            std::to_string(status) + ") at " + __FILE__ + ":" + std::to_string(__LINE__);
+        MIGRAPHX_THROW(EXIT_FAILURE, error_message);
+    }
+    return status;
+}
+
+template <class F, class Pack, class... Ts>
+auto hipblaslt_invoke(F f, Pack p, Ts... xs)
+{
+    return p([=](auto... ws) {
+        auto status = f(ws..., xs...);
+        if(status != HIPBLAS_STATUS_SUCCESS)
+        {
+            MIGRAPHX_THROW("hipblaslt_invoke: hipBlasLt call failed with status " +
+                           std::to_string(status));
+        }
+        return status;
+    });
+}
```

Comment:
These functions should be in the migraphx::gpu namespace.

Reply:
Done.